.circleci/bin/deploy-ci.py

```diff
@@ -0,0 +1,498 @@
+        if service in ["all", "identity"]:
+            deploy_list += IDENTITY_NODES
+
+    # make sure hosts is not a superset of deploy_list
+    for host in hosts:
+        if host not in deploy_list:
+            logger.error(f"'{host}' not found within service nodes for '{service}'")
+            logger.error("Did you mean to use both -s and -h?")
+            exit(1)
+
+    # only deploy the subset of hosts
+    if hosts:
+        deploy_list = hosts
+
+    return deploy_list
+
+
+def format_artifacts(heading=None, hosts=None, release_summary=None):
+    """Send summary output to stdout and /tmp/summary.md|json for artifact collection."""
+
+    # write summary.md
+    if hosts:
+        hosts.sort()
+        summary = [f"{heading}:"]
+        for h in hosts:
+            summary.append(f"* {h}")
+
+        print("\n".join(summary))
+        with open("/tmp/summary.md", "a") as f:
+            f.write("\n".join(summary))
+            f.write("\n\n")
+
+    # write summary.json
+    if release_summary:
+        with open("/tmp/summary.json", "w") as f:
+            f.write(json.dumps(release_summary, indent=4))
+
+
+@click.command()
+@click.option("-u", "--github-user", required=True)
+@click.option("-g", "--github-token", required=True)
+@click.option("-t", "--git-tag", required=False)
+@click.option(
+    "-f",
+    "--force",
+    is_flag=True,
+    default=False,
+    help="Deploy over nodes running feature branches",
+)
+@click.option(
+    "-e", "--environment", type=click.Choice(ENVIRONMENTS, case_sensitive=False)
+)
+@click.option(
+    "-s",
+    "--services",
+    type=click.Choice(SERVICES),
+    required=False,
+    default=("all",),
+    multiple=True,
+)
+@click.option(
+    "-h", "--hosts", type=click.Choice(ALL_NODES), required=False, multiple=True
+)
+@click.option(
+    "--parallel-mode/--no-parallel-mode",
+    " /-P",
+    show_default=True,
+    default=True,
+)
+@click.option(
+    "-d",
+    "--dry-run",
+    is_flag=True,
+    default=False,
+)
+def cli(
+    github_user,
+    github_token,
+    git_tag,
+    force,
+    environment,
+    services,
+    hosts,
+    parallel_mode,
+    dry_run,
+):
+    """
+    Deploy a git_tag (defaults to latest `master` commit when unspecified)
+    across a combination of: environments, services, and hosts.
+    """
+
+    if not git_tag:
+        # use the tip of `master`
+        git_tag = run_cmd("git log -n 1 --pretty=format:%H master")
+
+    # gather and display current release state, pre-deploy
+    release_summary = {
+        "deploy_list": generate_deploy_list(services, hosts),
+        "git_tag": git_tag,
+    }
+    update_release_summary(
+        release_summary,
+        PRE_DEPLOY,
+        parallel_mode=parallel_mode,
+        github_user=github_user,
+        github_token=github_token,
+        force=force,
+    )
+    print_release_summary(release_summary)
+
+    # add future logic to avoid multiple parallel deployments
+    seconds = 1
```

Comment:
can we remove this block for now? i don't really understand the intent?

Reply:
Fair point!